### ISSUE TITLE
Fails to build on 1.29.0 nightly

### ISSUE THREAD

**Issue by theindigamer** [rust-lang-nursery/rust-clippy#2900]
Running `cargo +nightly install clippy` gives

```
$ rustc --version
rustc 1.29.0-nightly (e06c87544 2018-07-06)

$ cargo +nightly install clippy
blah blah
[...] -- associated item not found in syntax::codemap::Spanned<..>
error: aborting due to 14 previous errors

Some errors occurred: E0223, E0432, E0531, E0599, E0609.
For more information about an error, try `rustc --explain E0223`.
error: failed to compile `clippy v0.0.211`, intermediate artifacts can be found at `/tmp/cargo-installgIHK4f`
```

Sorry if this is a dup, couldn't find this mentioned for 1.29.0.

**Comment by mikerite** [comment]
#2889 fixes this.
It just needs a maintainer to merge and update versions.

**Comment by Manishearth** [comment]
I'll fix and merge in a bit.

We just made clippy a rustup component (`rustup component add clippy-preview`, may need to wait a couple more hours for this to work) so there's going to be some churn.

**Comment by Manishearth** [comment]
Merged and published as 0.0.212

**Comment by theindigamer** [comment]
Thanks!
